### ISSUE TITLE
[Tier 1] In-container Python sanity: torch / aiter / flash_attn / TE import on gfx950

### ISSUE THREAD

**Issue by sunxxuns** [amdpilot-org/miles#4]
## Goal
Catch ABI / build-flag mismatches between the prebuilt wheels in the AMD
image and the gfx950 runtime before any training code runs.

## Steps
Inside the container, run:
```python
import torch, aiter, flash_attn
import transformer_engine.pytorch as te
print("torch", torch.__version__, "hip", torch.version.hip)
print("device count", torch.cuda.device_count())
print("aiter", aiter.__version__ if hasattr(aiter, "__version__") else "ok")
print("flash_attn", flash_attn.__version__)
print("TE", te.__version__)
x = torch.randn(8, 8, device="cuda"); print("matmul ok", (x @ x).sum().item())
```

## Acceptance
- All imports succeed without `ImportError` / `ROCmError` / `Symbol not found`.
- `torch.cuda.device_count() == 8`.
- A trivial matmul on device 0 returns a finite scalar.
- Failures must be captured verbatim — they are the input for the Tier-2 patch issues.
---
**Target arch**: gfx950 (MI355X)
**Base image**: `rlsys/miles:MI350-355-latest` (Docker Hub, pushed 2026-05-12, pulls cleanly)
**Test cluster**: n08-09 (MI355X), g46 (MI355X)
**Upstream parity**: this issue tracks behavior on amdpilot-org/miles, kept
in sync with `radixark/miles@main`. File a separate upstream issue only after
confirming repro on a clean checkout.

Parent: #2

**Comment by jhinpan** [comment]
**AMDPilot — accumulated progress on this issue**

_Auto-generated from successive trials. Latest update: 2026-05-15T10:43:02Z by `issue-miles-4`._

### Verified infra fixes (apply during Stage 0)

- Reuse existing formulation artifacts (task.yaml, test_harness.py, Dockerfile) to skip redundant generation steps
- Reused existing formulation artifacts (task.yaml, test_harness.py, Dockerfile) to skip redundant generation steps


### Working approaches and their best metric

_(none yet — when a trial measurably improves the metric, the change lands here.)_


### Approaches that did NOT help — don't retry these

- Trial 1 failed verification (metric=10.0, verified=0.0)
- Supervisor triggering Trial 2 immediately after Trial 1 start without waiting for recon_complete signal
- Waiting for 1835s hard timeout to detect zero agent_progress events instead of failing fast


### Open ideas to try next

- Add state check in TrialManager.start_next_trial to wait for recon_complete signal before launching next trial
- Implement liveness check in ExecutionMonitor to flag zero agent_progress events after 300s


_amdpilot bot. Edit between the markers to add/correct fixes; the next update will preserve your changes. To disable updates on this issue, replace the body with the marker line only._

**Comment by sunxxuns** [comment]
Closing per xsun (cycle 12 audit): this is verification/measurement work that doesn't require source-code or kernel changes — not worth burning agent + GPU lease overhead. The orchestrator is now restricted to issues that demand actual code work (kernel ports, HIP/MFMA rewrites, source patches). The remaining open issues (#18, #19, #23) are the real porting work.
